docs/mvc/tutorials/mvc-with-entity-framework/sample/src/ContosoBooks/Controllers/AuthorController.cs

```diff
@@ -0,0 +1,134 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Mvc;
+using ContosoBooks.Models;
+using Microsoft.Framework.Logging;
+using Microsoft.AspNet.Mvc.Rendering;
+using Microsoft.Data.Entity.Storage;
+using Microsoft.Data.Entity;
+
+// For more information on enabling MVC for empty projects, visit http://go.microsoft.com/fwlink/?LinkID=397860
+
+namespace ContosoBooks.Controllers
+{
+    public class AuthorController : Controller
+    {
+        [FromServices]
+        public BookContext DbContext { get; set; }
+
+        [FromServices]
+        public ILogger<AuthorController> Logger { get; set; }
+
+        public IActionResult Index()
+        {
+            return View(DbContext.Authors);
+        }
+
+        public async Task<ActionResult> Details(int id)
+        {
+            Author author = await DbContext.Authors
+                .SingleOrDefaultAsync(x => x.AuthorID == id);
+            if (author == null)
+            {
+                return HttpNotFound();
+            }
+            return View(author);
+        }
+
+        public ActionResult Create()
+        {
+            return View();
+        }
+
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public async Task<ActionResult> Create([Bind("LastName", "FirstMidName")] Author author)
+        {
+            try
+            {
+                if (ModelState.IsValid)
+                {
+                    DbContext.Authors.Add(author);
+                    await DbContext.SaveChangesAsync();
+                    return RedirectToAction("Index");
+                }
+            }
+            catch (DataStoreException ex)
+            {
+                Logger.LogError("Unable to save changes.", ex);
```

Comment:
The error will already be written to the log by EF... but fine if you want to leave it too :smile: